cabal-install/Distribution/Client/DistDirLayout.hs

```diff
@@ -107,14 +111,20 @@ data DistDirLayout = DistDirLayout {
      }
 
 
+data StoreDirLayout = StoreDirLayout {
```

Comment:
That idea is not really compatible with the notion of a nix style store. A nix style store really means putting all files for a package in their own dirs, and not having shared stuff. I realise we sort-of break that with the shared package db, but one can see it as a thing derived from the store.

Also, it's not clear that for the OSX limits that we must use a scheme with a single shared dir for all libs. It may well be possible to use a scheme with a single RPATH for the store and incorporate the store subdir names into the library install names.